odl/contrib/tomo/elekta.py

```diff
@@ -0,0 +1,198 @@
+    angles = np.array(angles, dtype=float)
+    detector_shape = np.array(detector_shape, dtype=int)
+
+    # Constant system parameters
+    pixel_size = 0.368
+    det_shape_mm = np.array([287.04, 264.96])
+
+    # Create a (possible non-uniform) partition given the angles
+    angle_partition = odl.nonuniform_partition(angles)
+
+    # Compute the detector partition
+    piercing_point_mm = pixel_size * piercing_point
+    det_min_pt = -piercing_point_mm
+    det_max_pt = det_min_pt + det_shape_mm
+    detector_partition = odl.uniform_partition(min_pt=det_min_pt,
+                                               max_pt=det_max_pt,
+                                               shape=detector_shape)
+
+    # Create the geometry
+    geometry = odl.tomo.CircularConeFlatGeometry(
+        angle_partition, detector_partition,
+        src_radius=sad, det_radius=sdd - sad,
+        axis=[0, 0, 1])
+
+    return geometry
+
+
+def elekta_icon_space(shape=(448, 448, 448), **kwargs):
+    """Default reconstruction space for the Elekta Icon CBCT.
+
+    See the [whitepaper]_ for further information.
+
+    Parameters
+    ----------
+    shape : sequence of int, optional
+        Shape of the space, in voxels.
+    kwargs :
+        Keyword arguments to pass to `uniform_discr` to modify the space, e.g.
+        use another backend.
+
+    Returns
+    -------
+    elekta_icon_space : `DiscreteLp`
+
+    Examples
+    --------
+    Create default space:
+
+    >>> from odl.contrib import tomo
+    >>> space = tomo.elekta_icon_space()
+
+    Create sub-sampled space:
+
+    >>> space = tomo.elekta_icon_space(shape=(100, 100, 100))
+
+    See Also
+    --------
+    elekta_icon_geometry: Geometry for the Elekta Icon CBCT.
+    elekta_icon_fbp: Default reconstruction method for the Elekta Icon CBCT.
+
+    References
+    ----------
+    .. [whitepaper] *Design and performance characteristics of a Cone Beam
+       CT system for Leksell Gamma Knife Icon*
+    """
+    return odl.uniform_discr(min_pt=[-112.0, -112.0, 0.0],
+                             max_pt=[112.0, 112.0, 224.0],
+                             shape=shape,
+                             dtype='float32')
+
+
+def elekta_icon_fbp(ray_transform,
+                    padding=False, filter_type='Hann', frequency_scaling=0.6,
+                    parker_weighting=True):
+    """Approximation of the FDK reconstruction used in the Elekta Icon.
+
+    Parameters
+    ----------
+    ray_transform : `RayTransform`
+        The ray transform to be used, should have an Elekta Icon geometry.
+    padding : bool, optional
+        If the fbp filter should use padding, increases memory use
+        significantly.
+    filter_type : str, optional
+        Type of filter to apply in the FBP filter.
+    frequency_scaling : float, optional
+        Frequency scaling for FBP filter
+    parker_weighting : bool, optional
+        True if parker weighting should be applied to compensate for finite
```

Comment:
Parker
finite? I guess you mean "less than 360 degrees", maybe "incomplete" or "less than full-circle".